code/go/0chain.net/smartcontract/zcnsc/nodes.go

```diff
@@ -347,8 +348,10 @@ func AuthorizerFromEvent(ev *event.Authorizer) (*AuthorizerNode, error) {
 // ----- UserNode ------------------
 
 type UserNode struct {
-	ID        string `json:"id"`
-	BurnNonce int64  `json:"burn_nonce"`
+	ID          string              `json:"id"`
+	BurnNonce   int64               `json:"burn_nonce"`
+	MintNonces  []int64             `json:"mint_nonces"`
+	BurnTickets map[string][][]byte `json:"burn_tickets"`
```

Comment:
The BurnTickets could become pretty large when user has lot of burned tickets, we should avoid storing data that could grow without limit in a single MPT node. So please refactor to store data with [partitions](https://github.com/0chain/0chain/tree/85f47d6f3d19eafa0465dad19c237aed5b5028ed/code/go/0chain.net/smartcontract/partitions) if we don't do it in this PR.

For the partitions refactor:
We need to have two kinds of partitions. One for storing the users burn tickets eth addresses so that we can list all eth addresses of a user when needed. We can use `${SCAddress}+Hash("burn_tickets_eth_addr:${client_id}")` as partititon name, the partition will store all the burn ticket eth address related to the client.

And use another partition to store burn tickets info, that use `${SCAddress}+Hash("burn_tickets:${client_id}")` as partition name, the partition will store all the burn tickets related to the client_id.

Reply:
> Let's use `BurnTickets as map[string]*entity.BurnTicket`, and run msgp to generate encode/decode code for the `entity.BurnTicket`. We should avoid doing json encode/decode ourself for MPT data.

If we use our own structure as a value type msgp won't generate encode/decode code for us, it will force us to implement our own encode/decode functions, but I will sort it out, how to improve it

Reply:
@YarikRevich we'd better define the `BurnTicket` struct in the `zcnsc` package rather than create a new package for it. And run `make msgp` in the project root will generate the encode/decode code for us.